ArcMovies/MoviesScene/CollectionView/Layout/MoviesFlowLayout.swift

```diff
@@ -0,0 +1,144 @@
+    }
+
+}
+
+// MARK: Overrides
+
+extension MoviesFlowLayout {
+    override func invalidateLayout(with context: UICollectionViewLayoutInvalidationContext) {
+        super.invalidateLayout(with: context)
+        guard let collectionView = self.collectionView else { return }
+
+        if collectionView.bounds.size != lastCollectionViewSize {
+            configureInsets()
+            lastCollectionViewSize = collectionView.bounds.size
+        }
+    }
+    
+    override func shouldInvalidateLayout(forBoundsChange _: CGRect) -> Bool {
+        return true
+    }
+    
+    override func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        guard let collectionView = self.collectionView,
+              let elementsArray = super.layoutAttributesForElements(in: rect) else {
+            return super.layoutAttributesForElements(in: rect)
+        }
+        
+        let size = collectionView.bounds.size
+        let contentOffset = collectionView.contentOffset
+
+        let visibleRect = CGRect(x: contentOffset.x, y: contentOffset.y, width: size.width, height: size.height)
+        let visibleCenterX = visibleRect.midX
+
+        elementsArray.forEach {
+            let distanceFromCenter = visibleCenterX - $0.center.x
+            let absDistanceFromCenter = min(abs(distanceFromCenter), scaleOffset)
+            let scale = absDistanceFromCenter * (self.minScaleFactor - 1) / self.scaleOffset + 1
+            $0.transform3D = CATransform3DScale(CATransform3DIdentity, scale, scale, 1)
+            
+            let alpha = absDistanceFromCenter * (self.minAlphaFactor - 1) / self.scaleOffset + 1
+            $0.alpha = alpha
+        }
+
+        return elementsArray
+    }
+    
+    override func targetContentOffset(forProposedContentOffset proposedContentOffset: CGPoint, withScrollingVelocity velocity: CGPoint) -> CGPoint {
+        guard let collectionView = self.collectionView else {
+            return proposedContentOffset
+        }
+        
+        let proposedRect = CGRect(x: proposedContentOffset.x, y: 0, width: collectionView.bounds.width, height: collectionView.bounds.height)
+        guard let layoutAttributes = self.layoutAttributesForElements(in: proposedRect) else {
+            return proposedContentOffset
+        }
+        
+        var candidateAttributes: UICollectionViewLayoutAttributes?
+        let proposedContentOffsetCenterX = proposedContentOffset.x + collectionView.bounds.width / 2
+        
+        for attributes in layoutAttributes {
+            if attributes.representedElementCategory != .cell {
+                continue
+            }
+            
+            if candidateAttributes == nil {
+                candidateAttributes = attributes
+                continue
+            }
+            
+            if fabs(attributes.center.x - proposedContentOffsetCenterX) < fabs(candidateAttributes!.center.x - proposedContentOffsetCenterX) {
```

Comment:
Why force unwrap?

Reply:
Because exclusively here, I have absolutely sure the candidateAttributes will not broke the code, because I'm already checking the if ```candidateAttributes == nil``` to break the current loop  iteration. But a better approach could be:

```
guard let unwrapCandidateAttributes = candidateAttributes else {
     candidateAttributes = attributes
     continue
 }
```